### PR TITLE
CLD-487: Add support for wallet password and realm in docker init

### PR DESCRIPTION
**Pull request by barkhachoithani**
- added env var support
- added docker test case

### CODE REVIEW

File: src/centos/scripts/start-marklogic.sh [hunk 1]
```diff
@@ -145,15 +155,29 @@ elif [[ "${MARKLOGIC_INIT}" == "true" ]]; then
         LICENSE_PAYLOAD="{\"license-key\" : \"${LICENSE_KEY}\",\"licensee\" : \"${LICENSEE}\"}"
     fi
 
-    log "Initialzing MarkLogic on ${HOSTNAME}."
```
[the remark this answers is not in the document]
Comment (barkhachoithani) [reply src/centos/scripts/start-marklogic.sh:148]: this was removed by mistake, added  it back.

File: src/centos/scripts/start-marklogic.sh [hunk 2]
```diff
@@ -129,6 +131,14 @@ else
     ML_ADMIN_USERNAME="${MARKLOGIC_ADMIN_USERNAME}"
 fi
 
+if [[ -f "$SECRET_WALLET_PWD_FILE" ]] && [[ -n "$(<"$SECRET_WALLET_PWD_FILE")" ]]; then
+    log "Using docker secrets for credentials."
```
[the remark this answers is not in the document]
Comment (barkhachoithani) [reply src/centos/scripts/start-marklogic.sh]: Updated the log message.

File: test/docker-test-cases.json [hunk 3]
```diff
@@ -63,6 +63,22 @@
             }
         }
     },
+    "docker-5": {
```
[the remark this answers is not in the document]
Comment (barkhachoithani) [reply test/docker-test-cases.json:66]: There is no API or admin UI functionality to test if pass-phrase for wallet was set with the value given in environment variable. It's the same for the admin password.
For realm, the user can specify their own value. I have tested setting realm to a different value than the public. It works as expected.

**Review by vitalykorolev — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -74,13 +74,15 @@ With this image, you have the option to either create an initialized or an unini
...
 
-To create an initialized MarkLogic Server, pass in environment variables and replace `{insert admin username}`/`{insert admin password}` with actual values for admin credentials. Optionally, you can pass license information in `{insert license}`/`{insert licensee}` to apply your MarkLogic license. To do this, run this this command: 
+To create an initialized MarkLogic Server, pass in environment variables MARKLOGIC_ADMIN_USERNAME, MARKLOGIC_ADMIN_PASSWORD and replace `{insert admin username}`/`{insert admin password}` with actual values for admin credentials. Use optional environment variable MARKLOGIC_WALLET_PASSWORD and REALM to set wallet password and authentication realm of the admin user. If not provided  wallet-password will default to value set for admin-password and realm will be set to public. Optionally, you can pass license information in `{insert license}`/`{insert licensee}` to apply your MarkLogic license. To do this, run this this command:
```
Comment (jmckean17) [inline README.md:77]: To create an initialized MarkLogic Server, pass in the environment variables MARKLOGIC_ADMIN_USERNAME and MARKLOGIC_ADMIN_PASSWORD, and replace `{insert admin username}`/`{insert admin password}` with actual values for admin credentials. Use the optional environment variable MARKLOGIC_WALLET_PASSWORD and REALM to set the wallet password and authentication realm of the admin user. If not provided, the wallet-password will default to the value set for admin-password and realm will be set to public.

(last sentence OK as is)

File: README.md [hunk 5]
```diff
@@ -231,12 +238,12 @@ You can change the number of HugePages available to each MarkLogic container by
 
 MarkLogic Server Docker containers ship with a small set of scripts, making it easy to create clusters. See the [MarkLogic documentation](https://docs.marklogic.com/guide/concepts/clustering) for more about clusters. The following three examples show how to create MarkLogic Server clusters with Docker containers. The first two use Docker compose scripts to create one-node and three-node clusters. See the documentation for [Docker compose](https://docs.docker.com/compose/) for more details. The third example demonstrates a container setup on separate VMs.
 
-The credentials for the admin user are configured using Docker secrets, and are stored in `mldb_admin_username.txt` and `mldb_admin_password.txt` files.
+The credentials for the admin user are configured using Docker secrets, and are stored in `mldb_admin_username.txt`, `mldb_admin_password.txt` and `mldb_wallet_password.txt` files.
```
Comment (jmckean17) [inline README.md:241]: add a comma after `mldb_admin_password.txt`  

 `mldb_admin_password.txt`, and `mldb_wallet_password.txt` files.

File: README.md [hunk 6]
```diff
@@ -231,12 +238,12 @@ You can change the number of HugePages available to each MarkLogic container by
...
 
 ## Single node MarkLogic Server on a single VM
 Single node configurations are used primarily on a development machine with a single user.
 
-Create these files on your host machine: `marklogic-centos.yaml`, `mldb_admin_username.txt`, and `mldb_admin_password.txt`. Run example Docker commands from the same directory where the files created. 
+Create these files on your host machine: `marklogic-centos.yaml`, `mldb_admin_username.txt`, `mldb_admin_password.txt` and `mldb_wallet_password.txt`. Run example Docker commands from the same directory where the files created.
```
Comment (jmckean17) [inline README.md:246]: add a comma after `mldb_admin_password.txt`,

Run the example Docker commands from the same directory that the files were created.

File: README.md [hunk 7]
```diff
@@ -431,7 +451,7 @@ As in the previous single-node example, each node of the cluster can be accessed
...
+In the previous examples, Docker secrets files were used to specify admin credentials for the MarkLogic Server. If your environment prevents the use of Docker secrets, you can use environmental variables. This approach is less secure, but it is commonly used in development environments. This is not recommended for production environments. In order to use these environment variables in the Docker compose files, remove the secrets section at the end of the Docker compose yaml file, and remove the secrets section in each node. Then replace the MARKLOGIC_ADMIN_USERNAME_FILE/MARKLOGIC_ADMIN_PASSWORD_FILE/MARKLOGIC_WALLET_PASSWORD_FILE variables with MARKLOGIC_ADMIN_USERNAME/MARKLOGIC_ADMIN_PASSWORD/MARKLOGIC_WALLET_PASSWORD and provide the appropriate values.
```
Comment (jmckean17) [inline README.md:454]: This is **not** recommended for production environments.

(can you make "not" bold?)

File: README.md [hunk 8]
```diff
@@ -553,29 +591,39 @@ Now that the nodes have been initialized, we rotate the secrets files to overwri
 ```
   $docker secret create mldb_admin_password_v2 mldb_admin_password_v2.txt
 ```
+- Create mldb_wallet_password_v2.txt and use the following command to add a new docker secret for the wallet password:
```
Comment (jmckean17) [inline README.md:594]: Create mldb_wallet_password_v2.txt and use the following command to add a new Docker secret for the wallet password:

(capitalize Docker)

File: README.md [hunk 9]
```diff
@@ -553,29 +591,39 @@ Now that the nodes have been initialized, we rotate the secrets files to overwri
...
+```
+  $docker secret create mldb_wallet_password_v2 mldb_wallet_password_v2.txt
+```
 6. Use the following commands to rotate the docker secrets for all the docker services created above using docker stack:
```
Comment (jmckean17) [inline README.md:598]: Use the following commands to rotate the Docker secrets for all the Docker services created above using Docker stack:

(capitalize Docker)

File: src/centos/scripts/start-marklogic.sh [hunk 10]
```diff
@@ -129,6 +132,14 @@ else
     ML_ADMIN_USERNAME="${MARKLOGIC_ADMIN_USERNAME}"
 fi
 
+if [[ -f "$SECRET_WALLET_PWD_FILE" ]] && [[ -n "$(<"$SECRET_WALLET_PWD_FILE")" ]]; then
+    log "Using docker secret for wallet-password."
```
Comment (jmckean17) [inline src/centos/scripts/start-marklogic.sh:136]: log "Using Docker secret for wallet-password."

(capitalize Docker)

**Review by jmckean17 — APPROVED** [review]
Just a couple of things... Thanks!

File: README.md (hunk 4 above)
Comment (barkhachoithani) [reply README.md]: Fixed.

File: README.md (hunk 5 above)
Comment (barkhachoithani) [reply README.md]: Fixed.

File: README.md (hunk 8 above)
Comment (barkhachoithani) [reply README.md]: Fixed.

File: src/centos/scripts/start-marklogic.sh (hunk 10 above)
Comment (barkhachoithani) [reply src/centos/scripts/start-marklogic.sh]: Fixed.

File: README.md (hunk 9 above)
Comment (barkhachoithani) [reply README.md]: Fixed.

File: README.md (hunk 6 above)
Comment (barkhachoithani) [reply README.md]: Fixed.

**Pull request merged**